python/GafferUI/MessageWidget.py

```diff
@@ -268,30 +267,6 @@ def handle( self, level, context, msg ) :
 
 				w._addMessage( level, context, msg )
```

Comment:
I think we can remove the `if QtCore.QThread.currentThread() == application.thread()` conditional as well, since `executeOnUIThread()` will do the shortcut for the main thread for us anyway.

Reply:
Done.  I haven't actually tested this last round of changes much ... we do have some test coverage, and the tests still pass, but do we have a good example that actually uses this stuff in a UI I should test?

Reply:
`gaffer dispatch -gui -tasks GafferDispatch.PythonCommand -show PythonCommand` with a PythonCommand that calls  `IECore.msg` and operates in immediate mode should be a decent way I think.